lib/trento/application/usecases/hosts/heartbeats.ex

```diff
@@ -28,15 +28,29 @@ defmodule Trento.Heartbeats do
         }
       )
 
-    Multi.new()
-    |> Multi.insert(:insert, changeset,
-      conflict_target: :agent_id,
-      on_conflict: {:replace, [:timestamp]}
-    )
-    |> Multi.run(:command, fn _, _ ->
-      dispatch_command(agent_id, :passing)
-    end)
-    |> Repo.transaction()
+    result =
+      Multi.new()
+      |> Multi.insert(:insert, changeset,
+        conflict_target: :agent_id,
+        on_conflict: {:replace, [:timestamp]}
+      )
+      |> Multi.run(:command, fn _, _ ->
+        dispatch_command(agent_id, :passing)
+      end)
+      |> Repo.transaction()
+
+    case result do
+      {:ok, _} ->
+        :ok
+
+      {:error, :command, :host_not_registered, _} ->
+        {:error, :not_found}
+
+      error ->
+        Logger.error("Error while updating heartbeat for agent #{agent_id}", error: inspect(error))
```

Comment:
Nothing wrong here, but i see that we use the error metadata here, but in every other place we include the error in the main message text

Reply:
ah yes, we did not decided yet where and when to use metadata so I will add the error to the message and defer this decision